File: backend/src/main/java/com/example/backend/entity/Bill.java
package com.example.backend.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "bills")
public class Bill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "学员不能为空")
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @Column(name = "enrollment_id")
    private Long enrollmentId;

    @Enumerated(EnumType.STRING)
    @Column(name = "bill_type", nullable = false, length = 20)
    private BillType billType;

    @NotNull(message = "账单金额不能为空")
    @DecimalMin(value = "0.0", message = "金额不能为负数")
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal amount;

    @Column(length = 255)
    private String description;

    @Column(name = "due_date")
    private LocalDate dueDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Status status = Status.UNPAID;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "payment_id")
    private Payment payment;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum BillType {
        TUITION, REGISTRATION, OTHER
    }

    public enum Status {
        UNPAID, PAID, OVERDUE
    }
}

File: frontend/src/types/bill.ts
import { Student } from './student'
import { Payment } from './payment'

export enum BillType {
  TUITION = 'TUITION',
  REGISTRATION = 'REGISTRATION',
  OTHER = 'OTHER'
}

export enum BillStatus {
  UNPAID = 'UNPAID',
  PAID = 'PAID',
  OVERDUE = 'OVERDUE'
}

export interface Bill {
  id: number
  student: Student
  enrollmentId?: number
  billType: BillType
  amount: number
  description?: string
  dueDate?: string
  status: BillStatus
  payment?: Payment
  createdAt: string
  updatedAt: string
}

export interface BillRequest {
  studentId: number
  billType: BillType
  amount: number
  description?: string
  dueDate?: string
}
